### DISCUSSION TITLE
Is basecalling during minknow run mandatory?

### DISCUSSION THREAD

**Discussion by gokalpcelik** [LooseLab/readfish#348] [Q&A] [answered]
Hi devs. 
I am working on a small setup to run minion and readfish to capture reads from certain regions. My question is about basecalling during minknow run. 

My setup only has a single gpu with limited vram (6GB). Is it possible to run the sequencing without basecalling and let only readfish use basecaller to pick reads? Can I do post-basecalling on POD5 files generated at the end of the run once adaptive_sampling is finished?

Regards.

**Comment by Adoni5** [comment, marked as answer]
Hi @gokalpcelik - that is absolutely possible yes.

If you don't turn on FASTQ/bam output when setting up your run in MinKNOW, only the `pod5` signal files are written out.
You can then basecall these after the run.

I hope that helps, let me know if not.
Rory

**Reply by gokalpcelik** [reply]
Thank you for the prompt reply. This tool is really awesome. 
By the way although the dorado plugin branch is not ready for primetime I realized that if I just drop in dorado.py code as guppy.py code with proper library installs into the 2023.1.1 release it works with the latest 7.3.9 dorado and minknow installation on 22.04 and 20.04. Do you devs think of any possible pitfalls of doing this kind of tinkering?

**Reply by Adoni5** [reply]
Obviously, be careful when doing something like that - I would strongly recommend testing with playback as described in the [README](https://github.com/LooseLab/readfish?tab=readme-ov-file#testing-base-calling-and-mapping).

However, what you are describing should work! You could also try installing locally from the `dorado_plugin` branch and specifying `caller.dorado` in your toml file. 

With your desired python environment active - 
```bash
git clone https://github.com/LooseLab/readfish.git
cd readfish
git checkout feature/duplex
pip install .
```
Matt has been working on it since I last touched it so I can't say for certain, but it _should_ work?

**Reply by mattloose** [reply]
It should work. You should run a test using playback first to make sure there are no unexpected crashes. We have not done a long run yet to test it.

**Reply by gokalpcelik** [reply]
I did a bunch of tests during the weekend with the playback so my experience is pretty much on that end. 

- If dorado.py is renamed and put into 2023.1.1 it works. 

- If 2024 version is installed from git it fails within a few minutes of run starting consistently. 

I have not tested the duplex branch but sure I will test it. 

I am using a minimal PC for minion mk1b with a Geforce GTX 1060 6gb gpu. My resource monitor indicated that vram usage never exceeded 3GBs even during basecalling enabled by both readfish and minknow. Of course there is only a single instance of dorado running so the socket is shared between 2 processes. I will install a second gpu of 2060 Super soon. Might have more room to improve performance I suppose. 

Looking at my notes I also tested a totally weird but working case of running an instance of guppy basecaller (still installable from the ont repos.) and dorado basecaller at the same time on the same gpu. VRAM usage is around 3.5 GB and guppy is used by readfish default install and dorado is used by minknow. So looks like the system can be quite flexible.

**Reply by mattloose** [reply]
Could you try the latest branch that was pushed on Saturday evening? That should fix the crach of a run failing within a few minutes.

**Reply by gokalpcelik** [reply]
Hi again. 
Branch updated on Saturday works fine now.
